onnxruntime/core/providers/nnapi/nnapi_builtin/nnapi_api_helper.h

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include "core/providers/nnapi/nnapi_builtin/nnapi_lib/NeuralNetworksWrapper.h"
+
+#ifdef __ANDROID__
+#include <android/api-level.h>
+#endif
+
+
+namespace onnxruntime {
+namespace nnapi {
+struct NnApi;
+class ModelBuilder;
+
+struct DeviceWrapper {
+  ANeuralNetworksDevice* device;
+  std::string name;
+  int32_t type;
+  int64_t feature_level;
+};
+
+enum class TargetDeviceOption : int8_t {
+  ALL_DEVICES,  // use all available target devices
+
+  /* TODO support these options
+  PREFERRED_DEVICES,  // Use one or more preferred devices (must be given)
+  EXCLUDED_DEVICES,   // Exclude one or more devices (must be given)
+   */
+
+  CPU_DISABLED,  // use all available target devices except CPU
+  CPU_ONLY,      // use CPU only
+};
+
+const char* const nnapi_cpu = ("nnapi-reference");
```

Comment:
We can use it in logs or somewhere and it's more concise to represent on where nnapi fallback

Reply:
I see that it could be useful in logging, but it doesn't seem to be used now as far as I can tell.

maybe here?
https://github.com/microsoft/onnxruntime/blob/9f26a0ef51dca22761cb4380c6d151535acca002/onnxruntime/core/providers/nnapi/nnapi_builtin/builders/model_builder.cc#L589

naming suggestion (see https://google.github.io/styleguide/cppguide.html#Constant_Names)

```suggestion
constexpr const char* kNnapiCpuDeviceName = "nnapi-reference";
```

Reply:
yeah, there is only one reference now.